File: src/cloudflare/durable_object.rs
use worker::*;

pub fn get_durable_stub(env: &Env, region: Option<&str>) -> Result<Stub> {
    let namespace = env.durable_object("MEKHALA_NWC_DO")?;
    match region {
        Some(r) if !r.is_empty() => namespace.get_by_name_with_location_hint("GLOBAL", r),
        _ => namespace.id_from_name("GLOBAL")?.get_stub(),
    }
}

File: src/cloudflare/durable_object.ts
export function getDurableStub(
  env: Record<string, unknown>,
  region?: string,
): DurableObjectStub {
  const ns = env.MEKHALA_NWC_DO as unknown as DurableObjectNamespace;
  if (region && region !== '') {
    return ns.getByName('GLOBAL', { locationHint: region as DurableObjectLocationHint });
  }
  const id = ns.idFromName('GLOBAL');
  return ns.get(id);
}
